bin/millau/runtime/src/rialto.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Configuration parameters for the Rialto Substrate chain.
+
+use bp_rialto::Header;
+use hex_literal::hex;
+use pallet_substrate_bridge::AuthoritySet;
+use sp_core::crypto::Public;
+use sp_finality_grandpa::AuthorityId;
+use sp_std::vec;
+
+/// The first header known to the pallet.
+///
+/// Note that this does not need to be the genesis header of the Rialto
+/// chain since the pallet may start at any arbitrary header.
+// To get this we first need to call the `chain_getBlockHash` RPC method, and then
+// we can use the result from that and call the `chain_getBlock` RPC method to get
+// the rest of the info.
+//
+// In this case we've grabbed the genesis block of the Rialto Substrate chain.
+pub fn initial_header() -> Header {
+	Header {
+		parent_hash: Default::default(),
+		number: Default::default(),
+		state_root: hex!("8fc2e3a8216b9f70fe2bc6804febb3f5192cb4046db5d546d1ad499a29bbbf27").into(),
```

Comment:
I was getting different values locally && then realized we have a circular dependency here. Right now you have `Default::default()` in `rialto-runtime/src/millau.rs` && you may guess `state_root` of Rialto genesis block. But once you fix `state_root` in `rialto-runtime/src/millau.rs`, the `Rialto::genesis::state_root` would change and you will need to update it in this file => `Millau::genesis::state_root` will change and so on.

What we need is a separate 'start-headers-sync' step similar to `eth-deploy-contract` we have in sub2eth sync. So my suggestion is to:
1) since we have started with Millau -> Rialto headers sync (see PRs list), let's:
1.1) change `state_root` to `Default::default()` in this file;
1.2) change `state_root` from `Default::default()` to actual value in `bin/rialto/runtime/src/millau.rs` (this would allow us to start Millau -> Rialto headers and messages relay);
2) in next PRs change substrate pallet to:
2.1) support starting with empty configuration - i.e. no `initial_header`, no `initial_genesis_set`, ...;
2.2) support "initialize" sudo-call that will accept `GenesisConfig`
3) add additional function to relay - `initialize_rialto_headers_to_millau` that will submit tx with that sudo call.

Reply:
I opened  #407 to track this.